File: Java/GestionnaireCommandes.java
import java.util.ArrayList;
import java.util.List;

public class GestionnaireCommandes {
    private List<Commande> commandes;

    public GestionnaireCommandes() {
        this.commandes = new ArrayList<>();
    }

    public void ajouterCommande(Commande commande) {
        this.commandes.add(commande);
    }

    public void annulerCommande(int id) {
        this.commandes.removeIf(commande -> commande.getId() == id);
    }

    public void afficherCommandes() {
        this.commandes.forEach(Commande::afficherDetails);
    }

    public void modifierStatut(int idCommande, String nouveauStatut) {
        for (Commande commande : this.commandes) {
            if (commande.getId() == idCommande) {
                commande.modifierStatut(nouveauStatut);
                System.out.println("Statut de la commande mis à jour avec succès");
                return;
            }
        }
        System.out.println("Commande non trouvée");
    }

    public void afficherDetailsCommande(int idCommande) {
        for (Commande commande : this.commandes) {
            if (commande.getId() == idCommande) {
                commande.afficherDetails();
                return;
            }
        }
        System.out.println("Commande non trouvée");
    }

    public List<Commande> getCommandes() {
        return commandes;
    }
}

File: Python/GestionnaireCommandes.py
from Commande import Commande

class GestionnaireCommandes:
    def __init__(self):
        self.commandes = []

    def ajouter_commande(self, commande):
        self.commandes.append(commande)

    def annuler_commande(self, id):
        self.commandes = [commande for commande in self.commandes if commande.id != id]

    def afficher_commandes(self):
        for commande in self.commandes:
            commande.afficher_details()

    def modifier_statut(self, id_commande, nouveau_statut):
        for commande in self.commandes:
            if commande.id == id_commande:
                commande.modifier_statut(nouveau_statut)
                print("Statut de la commande mis à jour avec succès")
                return
        print("Commande non trouvée")

    def afficher_details_commande(self, id_commande):
        for commande in self.commandes:
            if commande.id == id_commande:
                commande.afficher_details()
                return
        print("Commande non trouvée")